### PR TITLE
fc: give every microVM its own guest hostname

### PR DESCRIPTION
**Pull request by cdalar**
## Summary
- Every Firecracker VM onctl creates booted with the same guest OS hostname (`onctl-runner`), regardless of its actual VM name — because `bake-fc-image.sh`'s debootstrap runs in a chroot that can't isolate the UTS namespace, so it bakes one fixed placeholder hostname into the base rootfs image, and nothing at VM-creation time overrode it per-VM.
- `DebugfsRootfsPreparer.Prepare` now takes a `hostname` parameter and, when non-empty, rewrites `/etc/hostname` in the per-VM writable rootfs copy — same "edit the ext4 image via `debugfs -w`, no mount, no root needed" technique already used for SSH-key injection. Shared the "build script, run `debugfs -w`" plumbing into a new `runDebugfsScript` helper.
- `ProviderFC.Deploy` passes `server.Name` through a new `sanitizeGuestHostname` (lowercase, `[a-z0-9-]` only, 63-byte RFC 1123 cap) before handing it to `Prepare` — needed because arbitrary VM names aren't guaranteed hostname-safe (`fcTapName`/`fcMAC` already hash the name for the same reason), and because this value is written into a debugfs script verbatim, so an unsanitized name could otherwise inject extra script commands.
- Every microVM (whatever created it — the onctl-runners controller's runner pool, `onctl create` run by hand, the sandboxes feature, etc.) now gets a guest hostname matching its actual VM name.

## Test plan
- [x] `make` (build + `go mod tidy` + `go fmt`) — clean
- [x] `go test ./pkg/cloud/... ./internal/providerfc/...` — all pass, including new tests: `TestDebugfsRootfsPreparer_Prepare_InjectsHostname`, `_NoHostname`, `_HostnameDebugfsFails`, and `TestSanitizeGuestHostname` (charset/case/length edge cases)
- [x] `golangci-lint run ./pkg/cloud/... ./internal/providerfc/...` — 0 issues
- [x] `go vet ./...`, `gofmt -l` on changed files — clean
- Note: `go test ./cmd/...` has 3 pre-existing failures (`TestReadConfig_NoConfigDirectory`, `TestActionCmd_DownloadFile_404`, `TestCreateFlagsBindToViper`) reproduced identically on unmodified `origin/main` — unrelated to this change, not touched by this PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01TB8wdxuaLyoYZBkfcD1vzv

### CODE REVIEW

File: pkg/cloud/fc.go [hunk 1]
```diff
@@ -335,6 +340,33 @@ func fcMAC(vmName string) string {
...
+		switch {
+		case r >= 'a' && r <= 'z', r >= '0' && r <= '9', r == '-':
+			b.WriteRune(r)
+		default:
+			b.WriteRune('-')
+		}
+	}
+	s := strings.Trim(b.String(), "-")
+	if len(s) > 63 {
+		s = strings.Trim(s[:63], "-")
```
[the remark this answers is not in the document]
Comment (cdalar) [reply pkg/cloud/fc.go]: Good catch — fixed in 1fa5bf0. When the sanitized name overflows 63 bytes, `sanitizeGuestHostname` now truncates to 54 bytes and appends `-` + an 8-hex-char MD5 hash of the full original name, so two names sharing a long common prefix no longer collide on the same guest hostname. Added `TestSanitizeGuestHostname_TruncationPreservesUniqueness` to cover it.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by cdalar** [comment]
`Build (stable)` failed on 1fa5bf0, but it's not this PR's failure: `github.com/cdalar/onctl/cmd` failed only on `TestActionCmd_DownloadFile_404` (cmd/coverage_test.go:153), which hits the live external endpoint `https://httpstat.us/404` and expects a 404 — this run got a 502 Bad Gateway from that service instead. Nothing in this PR touches `cmd/`; this diff only changes `pkg/cloud/fc.go` and its tests. This matches one of the 3 pre-existing failures already called out in the PR description as reproducing identically on unmodified `main`. The job has already auto-requeued on this commit; watching for it to go green.

---
_Generated by [Claude Code](https://claude.ai/code)_
